immuneML/reports/data_reports/PDBDataReport.py

```diff
@@ -0,0 +1,94 @@
+from immuneML.data_model.repertoire.Repertoire import Repertoire
+from immuneML.reports.ReportOutput import ReportOutput
+from immuneML.reports.ReportResult import ReportResult
+from immuneML.reports.data_reports.DataReport import DataReport
+from immuneML.util.PathBuilder import PathBuilder
+
+
+class PDBDataReport(EncodingReport):
+    """
+    Generates a histogram of the lengths of the sequences in a RepertoireDataset.
+
+    YAML specification:
+
+    .. indent with spaces
+    .. code-block:: yaml
+
+        my_sld_report: NewSequenceLengthDistribution
+
+    """
+
+    @classmethod
+    def build_object(cls,
+                     **kwargs):  # called when parsing YAML - all checks for parameters (if any) should be in this function
+        return PDBDataReport(**kwargs)
+
+    def __init__(self, dataset: RepertoireDataset = None, batch_size: int = 1, result_path: Path = None,
+                 name: str = None, randomInt: int = 0):
+        super().__init__(dataset=dataset, result_path=result_path, name=name)
+        self.randomInt = randomInt
+        self.batch_size = batch_size
+
+    def check_prerequisites(
+            self):  # called at runtime to check if the report can be run with params assigned at runtime (e.g., dataset is set at runtime)
+
+        return True
+
+    def _generate(self) -> ReportResult:  # the function that creates the report
+
+        dataset = np.array(self.dataset.encoded_data.examples.copy(),dtype=float)
+        output_files = []
+
+
+        for i in range(0, len(dataset)):
+
+                fileName = Path(self.dataset.get_example_ids()[i])
+                PathBuilder.build(self.result_path)
+
+                lightChainDistance = self.result_path / "{}_lightChainDistanceToAntigen.html".format(fileName)
+                heavyChainDistance = self.result_path / "{}_heavyChainDistanceToAntigen.html".format(fileName)
+
+                lightChainDistanceToAntigen = removeEmptyRows(removeInfValues(dataset[i][0]))
+                heavyChainDistanceToAntigen = removeEmptyRows(removeInfValues(dataset[i][1]))
+
+                figure = px.imshow(lightChainDistanceToAntigen)
+                figure.write_html(str(lightChainDistance))
+
+                fig2 = px.imshow(heavyChainDistanceToAntigen)
+                fig2.write_html(str(heavyChainDistance))
+
+
+
+                output_files.append(ReportOutput(
+                    name="{} - Heatmap of distance from carbon alphas in the LIGHT chain to carbon alphas in the antigen ".format(
+                        fileName), path=lightChainDistance))
+                output_files.append(ReportOutput(
+                    name="{} - Heatmap of distance from carbon alphas in the HEAVY chain to carbon alphas in the antigen ".format(
+                        fileName), path=heavyChainDistance))
+
+
+        return ReportResult(type(self).__name__, info="Heatmap", output_figures=output_files)
+
+
+def removeInfValues(arr):
```

Comment:
functions should be methods

Reply:
I actually didn't know there was a difference in the placement of the "def" keyword. It seems to me that functions are like public methods in Java, and methods are like private methods.

Reply:
in python indentation has meaning (unlike in some other languages where it is only used for readability). We do use functions in very rare cases in immuneML, but 99% of the time the immuneML 'functions' are methods, and thus inside a class. And we never mix classes and stand-alone functions in the same file. 
To distinguish between private and public methods we use the following naming convention: private method names start with an _underscore, whereas public methods don't. There is nothing that enforces private/public methods like in Java, so this naming convention is purely for ourselves for the sake of readability.

Reply:
(and in case it was not clear yet: function is stand-alone, method is class-bound)